pkg/compare/secret_key_reference.go

```diff
@@ -0,0 +1,103 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package compare
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+)
+
+// SecretKeyReferenceEqual returns true if the supplied secret key references
+// are equal.
+func SecretKeyReferenceEqual(a, b *ackv1alpha1.SecretKeyReference) bool {
+	if HasNilDifference(a, b) {
+		return false
+	}
+	return a.Name == b.Name &&
+		a.Namespace == b.Namespace &&
+		a.Key == b.Key
+}
+
+// SliceSecretKeyReferenceEqual returns true if the supplied slices of secret
+// key reference pointers contain the exact same elements.
+func SliceSecretKeyReferenceEqual(
+	a, b []*ackv1alpha1.SecretKeyReference,
+) bool {
+	equal, _, _ := compareSecretKeyReferenceEqual(a, b)
+	return equal
+}
+
+// compareSecretKeyReferenceEqual returns true if the supplied slices of secret
+// key reference pointers contain the exact same elements. It will also return
+// 2 slices containing the added and removed elements. The comparison doesn't
```

Comment:
nit: Maybe add `It will also return 2 slices containing elements contained in a that aren't in b, and elements contained in b that aren't in a, respectively.`

"added and removed" depends on the order of elements, so this would just clear up the confusion.